src/GafferSceneUI/LightTool.cpp

```diff
@@ -471,6 +471,74 @@ IECoreGL::MeshPrimitivePtr unitCone()
 	return result;
 }
 
+IECoreGL::MeshPrimitivePtr torus( const float width, const float height, const float tubeRadius, const Handle *handle, const Axis axis )
+{
+	IECoreGL::MeshPrimitivePtr result;
+
+	IECore::IntVectorDataPtr verticesPerFaceData = new IECore::IntVectorData;
+	std::vector<int> &verticesPerFace = verticesPerFaceData->writable();
+
+	IECore::IntVectorDataPtr vertexIdsData = new IECore::IntVectorData;
+	std::vector<int> &vertexIds = vertexIdsData->writable();
+
+	IECore::V3fVectorDataPtr pData = new IECore::V3fVectorData;
+	std::vector<V3f> &p = pData->writable();
+
+	const V3f radiusScale =
+		axis == Axis::X ? V3f( 0, height, width ) : (
+			axis == Axis::Y ? V3f( width, 0, height ) :
+			V3f( width, height, 0 )
+		)
+	;
+
+	const int numDivisionsI = 60;
+	const int numDivisionsJ = 15;
+	for( int i = 0; i < numDivisionsI; ++i )
+	{
+		const float iAngle = 2 * M_PI * (float)i / (float)( numDivisionsI - 1 );
+		const V3f v =
+			axis == Axis::X ? V3f( 0, cos( iAngle ), -sin( iAngle ) ) : (
+				axis == Axis::Y ? V3f( -sin( iAngle ), 0, cos( iAngle ) ) :
+				V3f( -sin( iAngle ), cos( iAngle ), 0 )
+			)
+		;
+		const V3f tubeCenter = v * radiusScale;
+
+		const int ii = i == numDivisionsI - 1 ? 0 : i + 1;
+
+		const float jRadius = tubeRadius * rasterScaleFactor( handle, tubeCenter );
+
+		for( int j = 0; j < numDivisionsJ; ++j )
+		{
+			const float jAngle = 2 * M_PI * (float)j / (float)( numDivisionsJ - 1 );
+			const V3f offset =
+				axis == Axis::X ? V3f( sin( jAngle ), 0, 0 ) : (
+					axis == Axis::Y ? V3f( 0, sin( jAngle ), 0 ) :
+					V3f( 0, 0, sin( jAngle ) )
+				)
+			;
```

Comment:
Since my job here is math code nitpicks, I'll note that there are 3 places here where you're switching based on `axis`. I would probably find it a bit clearer if you just picked a canonical orientation ( ie Axis::X ), and just kept the first branch of the 3 switches, computing everything in a local space, and then converted to your target space once at the end, ie something like:
```
V3f l = tubeCenter + jRadius * ( cos( jAngle ) * v + offset );
p.push_back( Axis::X ? V3f( l.x, l.y, l.z ) : ( axis == Axis::Y ? V3f( l.z, l.x, l.y ) : V3f( l.z, l.y, l.x ) ) );
```

Reply:
Maybe a little inline helper function so we don't even have to repeat the switch?

Reply:
Keeping everything in one orientation and converting once is definitely clearer. I combined that with the helper function and was able to improve the `circle` function a bit too. Those changes are in https://github.com/GafferHQ/gaffer/pull/5483/commits/5b30e52f8918a44b939bd90e88879fc84ab708f0